x-pack/plugins/profiling/public/components/differential_topn_functions_grid/index.tsx

```diff
@@ -0,0 +1,492 @@
+          <CPULabelWithHint
+            type="self"
+            labelSize="s"
+            labelStyle={{ fontWeight: 700 }}
+            iconSize="s"
+          />
+        ),
+      },
+      {
+        id: TopNFunctionSortField.TotalCPU,
+        actions: { showHide: false },
+        schema: 'numeric',
+        initialWidth: 120,
+        display: (
+          <CPULabelWithHint
+            type="total"
+            labelSize="s"
+            labelStyle={{ fontWeight: 700 }}
+            iconSize="s"
+          />
+        ),
+      },
+      {
+        id: TopNComparisonFunctionSortField.ComparisonRank,
+        actions: { showHide: false },
+        schema: 'numeric',
+        displayAsText: 'Rank',
+        initialWidth: 69,
+        displayHeaderCellProps: { className: 'thickBorderLeft' },
+      },
+      {
+        id: TopNComparisonFunctionSortField.ComparisonFrame,
+        actions: { showHide: false },
+        displayAsText: i18n.translate('xpack.profiling.functionsView.functionColumnLabel', {
+          defaultMessage: 'Function',
+        }),
+        cellActions: [compareFrameAction],
+      },
+      {
+        id: TopNComparisonFunctionSortField.ComparisonSamples,
+        actions: { showHide: false },
+        schema: 'numeric',
+        initialWidth: 120,
+        display: (
+          <LabelWithHint
+            label={i18n.translate('xpack.profiling.functionsView.samplesColumnLabel', {
+              defaultMessage: 'Samples',
+            })}
+            hint={i18n.translate('xpack.profiling.functionsView.samplesColumnLabel.hint', {
+              defaultMessage: 'Estimated values',
+            })}
+            labelSize="s"
+            labelStyle={{ fontWeight: 700 }}
+            iconSize="s"
+          />
+        ),
+      },
+      {
+        id: TopNComparisonFunctionSortField.ComparisonSelfCPU,
+        actions: { showHide: false },
+        schema: 'numeric',
+        initialWidth: 120,
+        display: (
+          <CPULabelWithHint
+            type="self"
+            labelSize="s"
+            labelStyle={{ fontWeight: 700 }}
+            iconSize="s"
+          />
+        ),
+      },
+      {
+        id: TopNComparisonFunctionSortField.ComparisonTotalCPU,
+        actions: { showHide: false },
+        schema: 'numeric',
+        initialWidth: 120,
+        display: (
+          <CPULabelWithHint
+            type="total"
+            labelSize="s"
+            labelStyle={{ fontWeight: 700 }}
+            iconSize="s"
+          />
+        ),
+      },
+      {
+        displayAsText: 'Diff',
+        actions: { showHide: false },
+        id: TopNComparisonFunctionSortField.ComparisonDiff,
+        initialWidth: 70,
+        isSortable: false,
+      },
+    ],
+    [compareFrameAction]
+  );
+
+  const [visibleColumns, setVisibleColumns] = useState(columns.map(({ id }) => id));
+
+  function RenderCellValue({ rowIndex, columnId, setCellProps }: EuiDataGridCellValueElementProps) {
```

Comment:
Since this returns a component, should we name it as something like this?

```suggestion
  function CellRenderer({ rowIndex, columnId, setCellProps }: EuiDataGridCellValueElementProps) {
```